pkgs/development/libraries/webp-pixbuf-loader/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib, stdenv, fetchFromGitHub
+, meson, ninja, pkg-config, makeWrapper
+, gdk-pixbuf, libwebp
+}:
+let
+  moduleDir = gdk-pixbuf.moduleDir;
+
+  # turning lib/gdk-pixbuf-#.#/#.#.#/loaders into lib/gdk-pixbuf-#.#/#.#.#/loaders.cache
+  # removeSuffix is just in case moduleDir gets a trailing slash
+  loadersPath = (lib.strings.removeSuffix "/" gdk-pixbuf.moduleDir) + ".cache";
+in
+stdenv.mkDerivation rec {
+  pname = "webp-pixbuf-loader";
+  version = "0.0.6";
+
+  src = fetchFromGitHub {
+    owner = "aruiz";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-dcdydWYrXZJjo4FxJtvzGzrQLOs87/BmxshFZwsT2ws=";
+  };
+
+  nativeBuildInputs = [ meson ninja pkg-config makeWrapper ];
+  buildInputs = [ gdk-pixbuf libwebp ];
+
+  mesonFlags = [
+    "-Dgdk_pixbuf_query_loaders_path=${gdk-pixbuf.dev}/bin/gdk-pixbuf-query-loaders"
+    "-Dgdk_pixbuf_moduledir=${placeholder "out"}/${moduleDir}"
+  ];
+
+  # The .thumbnailer file calls gdk-pixbuf-thumbnailer incorrectly.
+  # First, it looks for it in this package's bin rather than the gdk-pixbuf bin.
+  # Second, it assumes gdk-pixbuf-thumbnailer can find the webp loader in the loaders.cache referenced by environment variable, breaking containment.
+  # So we replace it with a wrapped executable.
+  postPatch = ''
+    substituteInPlace webp-pixbuf.thumbnailer.in --replace @bindir@/gdk-pixbuf-thumbnailer $out/bin/webp-thumbnailer
+  '';
+  postInstall = ''
+    mkdir -p $out/bin
+    makeWrapper \
+      ${gdk-pixbuf}/bin/gdk-pixbuf-thumbnailer \
```

Comment:
Are you sure about this? I think makeWrapper takes absolute paths.
I get this message with the change:
```
Builder called die: Cannot wrap 'gdk-pixbuf-thumbnailer' because it is not an executable file
```

Reply:
Oh, I missread that line.